server/middlewares/auth.js

```diff
@@ -0,0 +1,17 @@
+const jwt = require("jsonwebtoken");
+
+module.exports = (req, res, next) => {
+    try {
+        const token = req.headers.authorization.split(" ")[1];
+        jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user)=>{
+            if(err){
+                return res.status(401).json({ message: "Authentication failed!" });
+            }
+            req.user = user;
+            console.log(user)
+        });
+        next();
+    } catch (error) {
```

Comment:
are you sure we need this **try catch** here ?

Reply:
I changed this to

```
if(req.headers.authorization && req.headers.authorization.split(" ").length === 2){...}
else{...}
```